src/EditorFeatures/CSharpTest/InvertLogical/InvertLogicalTests.cs

```diff
@@ -0,0 +1,229 @@
+@"class C
+{
+    void M(bool x, int a, int b, int c)
+    {
+        var c = !(a <= 10 [||]&&
+                  b >= 20 &&
+                  c == 30);
+    }
+}",
+@"class C
+{
+    void M(bool x, int a, int b, int c)
+    {
+        var c = a > 10 ||
+                  b < 20 ||
+                  c != 30;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInvertLogical)]
+        public async Task InvertMultiConditional1()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(int a, int b, int c)
+    {
+        var x = a > 10 [||]|| b < 20 || c == 30;
+    }
+}",
+@"class C
+{
+    void M(int a, int b, int c)
+    {
+        var x = !(a <= 10 && b >= 20 && c != 30);
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInvertLogical)]
+        public async Task InvertMultiConditional2()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(int a, int b, int c)
+    {
+        var x = a > 10 || b < 20 [||]|| c == 30;
+    }
+}",
+@"class C
+{
+    void M(int a, int b, int c)
+    {
+        var x = !(a <= 10 && b >= 20 && c != 30);
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInvertLogical)]
+        public async Task InvertMultiConditional3()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(int a, int b, int c)
+    {
+        var x = !(a <= 10 [||]&& b >= 20 && c != 30);
+    }
+}",
+@"class C
+{
+    void M(int a, int b, int c)
+    {
+        var x = a > 10 || b < 20 || c == 30;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInvertLogical)]
+        public async Task InvertMultiConditional4()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(int a, int b, int c)
+    {
+        var x = !(a <= 10 && b >= 20 [||]&& c != 30);
+    }
+}",
+@"class C
+{
+    void M(int a, int b, int c)
+    {
+        var x = a > 10 || b < 20 || c == 30;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInvertLogical)]
+        public async Task TestMissingOnShortCircuitAnd()
+        {
+            await TestMissingAsync(
+@"class C
+{
+    void M(bool x, int a, int b)
+    {
+        var c = a > 10 [||]& b < 20;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInvertLogical)]
+        public async Task TestMissingOnShortCircuitOr()
+        {
+            await TestMissingAsync(
+@"class C
+{
+    void M(bool x, int a, int b)
+    {
+        var c = a > 10 [||]| b < 20;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInvertLogical)]
+        public async Task TestSelectedOperator()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool x, int a, int b)
+    {
+        var c = a > 10 [||||] b < 20;
```

Comment:
>[||||] [](start = 23, length = 6)

I'm amazed this works :-)

Reply:
[|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||]